lib/ansible/plugins/action/script.py

```diff
@@ -96,6 +98,10 @@ def run(self, tmp=None, task_vars=None):
         env_dict = dict()
         env_string = self._compute_environment_string(env_dict)
         script_cmd = ' '.join([env_string, tmp_src, args])
+
+        if self._play_context.check_mode:
+            return dict(changed=True)
+
```

Comment:
Not sure you need this down here, if you just set `self._supports_check_mode = False` in `__init__` or in `run` before calling `super` the base class will already skip this task. In any case you already have `result` dict with other information you might want to return `result['changed'] = True; return result`

Reply:
I like doing it down here because it allows the module to support check mode, not just be skipped. That means you can run it in check mode and if there are `creates:` or `removes:` parameters, those get evaluated in check mode. Otherwise, it'll just report a change, like `script` always does by default, but not actually run the script.

I'll update the `results` dict rather than returning a new one.